File: src/nastyhacks.cpp
#include <iostream>

using namespace std;

int main()
{
    int n, r, e, c;
    cin >> n;

    for (int i = 0; i < n; i++)
    {
        cin >> r >> e >> c;
        if ((e - c) > r)
        {
            cout << "advertise";
        }
        else if ((e - c) < r)
        {
            cout << "do not advertise";
        }
        else
        {
            cout << "does not matter";
        }
    }
}

File: src/nastyhacks.lua
for _ = 1, io.read("*n") do
    r, e, c = io.read("*n", "*n", "*n")
    if e > r + c then
        print("advertise")
    elseif e < r + c then
        print("do not advertise")
    else
        print("does not matter")
    end
end
